go/verify.go

```diff
@@ -0,0 +1,532 @@
+	if pos != 0 {
+		result = verifier.checkIndirectOffset(pos)
+		if result {
+			pos = verifier.getIndirectOffset(pos)
+			result = verifier.checkVector(pos, SizeByte)
+			if result && verifyObjFunc != nil {
+				vectorLength := USizeT(verifier.readUOffsetT(verifier.Buf[pos:]))
+				// Buffer begins after vector length
+				vectorStart := pos + SizeUOffsetT
+				nestedBuffer := verifier.Buf[vectorStart : vectorStart + UOffsetT(vectorLength)]
+				nestedVerifier := &Verifier{Buf: nestedBuffer, Pos: 0, options: verifier.options}
+				// There is no iternal identifier - use empty one
+				var identifier []byte
+				result = nestedVerifier.checkBufferFromStart(identifier, 0, verifyObjFunc)
+			}
+		}
+	} else {
+		result = !required
+	}
+	return result
+}
+
+// Method verifies union object using verification function
+//
+// (this method is used internally by generated verification functions)
+func (verifier *Verifier) VerifyUnion(tablePos UOffsetT, typeIdVOffset VOffsetT, valueVOffset VOffsetT,
+	verifyObjFunc VerifyUnionFunc, required bool) bool {
+	var result bool
+
+	// Check the union type id
+	pos := verifier.getVOffset(tablePos, typeIdVOffset)
+	if pos != 0 {
+		result = verifier.checkAlignment(pos, SizeByte) && verifier.checkElement(pos, SizeByte)
+		if result {
+			// Take type id
+			typeId := GetByte(verifier.Buf[pos:])
+			// Check union data
+			pos = verifier.getVOffset(tablePos, valueVOffset)
+			if pos != 0 {
+				result = verifier.checkIndirectOffset(pos)
+				if result {
+					// Take data position and validate union structure
+					pos = verifier.getIndirectOffset(pos)
+					result = verifyObjFunc(verifier, typeId, pos)
+				}
+			} else {
+				// When value data is not present, allow union verification function to deal with illegal offset
+				result = verifyObjFunc(verifier, typeId, UOffsetT(len(verifier.Buf)))
+			}
+		}
+	} else {
+		result = !required
+	}
+	return result
+}
+
+// Method verifies flatbuffer data using generated Table verification function.
+// The data buffer is already provided when creating [Verifier] object (see [NewVerifier])
+//
+//   - identifier - the expected identifier of buffer data.
+//     When empty identifier is provided the identifier validation is skipped.
+//   - sizePrefixed - this flag should be true when buffer is prefixed with content size
+//   - verifyObjFunc - function to be used for verification. This function is generated by compiler and included in each table definition file with name "<Tablename>Verify"
+//
+// Example:
+//
+//	/* Verify Monster table. Ignore buffer name and assume buffer does not contain data length prefix */
+//	isValid := verifier.VerifyBuffer([]byte{}, false, MonsterVerify)
+//
+//	/* Verify Monster table. Buffer name is 'MONS' and contains data length prefix */
+//	isValid := verifier.VerifyBuffer("MONS", true, MonsterVerify)
+func (verifier *Verifier) VerifyBuffer(identifier []byte, sizePrefixed bool, verifyObjFunc VerifyTableFunc) bool {
```

Comment:
In other places, I've used the type `string` for the identifier. So I suggest to use `string` also in this code, for consistency.

Reply:
fixed